src/main/java/com/twentyonec/ItemsLogger/utils/Storage.java

```diff
@@ -0,0 +1,261 @@
+			storage = new Storage(config.getHostname(), config.getPort(), config.getUsername(),
+					config.getPassword(), config.getDatabase());
+		}
+
+		return storage;
+	}
+
+	private void connect() {
+
+		final String jdbcUrl = "jdbc:mysql://" + this.hostname + ":" + this.port + "/" + this.database;
+
+		HikariConfig config = new HikariConfig();
+
+		config.setPoolName("itemslogger");
+		config.setJdbcUrl(jdbcUrl);
+		config.setUsername(this.username);
+		config.setPassword(this.password);
+
+		config.addDataSourceProperty("cachePrepStmts", "true");
+		config.addDataSourceProperty("prepStmtCacheSize", "250");
+		config.addDataSourceProperty("prepStmtCacheSqlLimit", "2048");
+		config.addDataSourceProperty("useServerPrepStmts", "true");
+		config.addDataSourceProperty("rewriteBatchedStatements", "true");
+		config.addDataSourceProperty("maintainTimeStats", "false");
+
+		config.setMaximumPoolSize(10);
+		config.setMinimumIdle(10);
+		config.setIdleTimeout(300000);
+		config.setMaxLifetime(600000);
+		config.setConnectionTimeout(5000);
+		config.setInitializationFailTimeout(-1);
+
+		try {
+			this.dataSource = new HikariDataSource(config);
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+
+	}
+
+	public Connection getConnection() throws SQLException {
+		return this.dataSource.getConnection();
+	}
+
+	public void update(final String sql) {
+
+		try (Connection connection = this.getConnection()) {
+
+			PreparedStatement statement = connection.prepareStatement(sql);
+
+			plugin.debugMessage("Preparing statement for update.");
+			statement.executeUpdate();
+			plugin.debugMessage("Successfully executed update statement. (" + sql + ")");
+
+		} catch (SQLException e) {
+			plugin.debugMessage("Error while attempting to execute update statement. (" + sql + ")");
+			e.printStackTrace();
+		}
+
+	}
+
+	public void update(String sql, Date date, Time time) {
+
+		try (Connection connection = this.getConnection()) {
+
+			PreparedStatement statement = connection.prepareStatement(sql);
+			statement.setDate(1, date);
+			statement.setTime(2, time);
+
+			plugin.debugMessage("Preparing statement for update.");
+			statement.executeUpdate();
+			plugin.debugMessage("Successfully executed update statement. (" + sql + ")");
+
+		} catch (SQLException e) {
+			plugin.debugMessage("Error while attempting to execute update statement. (" + sql + ")");
+			e.printStackTrace();
+		}
+
+	}
+
+	public ResultSet query(final String sql) {
+
+		try (Connection connection = this.getConnection()){
+
+			PreparedStatement statement = connection.prepareStatement(sql);
+
+			plugin.debugMessage("Preparing statement for query.");
+			ResultSet resultSet = statement.executeQuery(sql);
+			plugin.debugMessage("Successfully executed query statement. (" + sql + ")");
+
+			return resultSet;
+		} catch (SQLException e) {
+			plugin.debugMessage("Error while attempting to execute query statement. (" + sql + ")");
+			e.printStackTrace();
+			return null;
+		}
+	}
```

Comment:
Fix these methods to take strings with `?` placeholders and an array of inputs or something. Just don't do what you're doing right now